rllib/examples/quadx_waypoints.py

```diff
@@ -0,0 +1,89 @@
+import argparse
+import os
+
+from ray.rllib.examples.envs.classes.pyflyt_quadx_waypoints_env import (
+    create_quadx_waypoints_env,
+)
+from ray.rllib.utils.test_utils import check_learning_achieved
+from ray.tune.registry import get_trainable_cls
+
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    "--run", type=str, default="PPO", help="The RLlib-registered algorithm to use."
+)
+parser.add_argument("--num-cpus", type=int, default=4)
+parser.add_argument(
+    "--framework",
+    choices=["tf", "tf2", "torch"],
+    default="torch",
+    help="The DL framework specifier.",
+)
+parser.add_argument("--use-prev-action", action="store_true")
+parser.add_argument("--use-prev-reward", action="store_true")
+parser.add_argument(
+    "--as-test",
+    action="store_true",
+    help="Whether this script should be run as a test: --stop-reward must "
+    "be achieved within --stop-timesteps AND --stop-iters.",
+)
+parser.add_argument(
+    "--stop-iters", type=int, default=500, help="Number of iterations to train."
+)
+parser.add_argument(
+    "--stop-timesteps", type=int, default=1000000, help="Number of timesteps to train."
+)
+parser.add_argument(
+    "--stop-reward", type=float, default=90.0, help="Reward at which we stop training."
+)
+
+if __name__ == "__main__":
+    import ray
+    from ray import air, tune
+    from ray.tune.registry import register_env
+
+    args = parser.parse_args()
+
+    ray.init()
+
+    register_env("quadx_waypoints", create_quadx_waypoints_env)
+
+    algo_cls = get_trainable_cls(args.run)
+    config = algo_cls.get_default_config()
+
+    config.environment(env="quadx_waypoints").resources(
+        num_gpus=int(os.environ.get("RLLIB_NUM_GPUS", "0"))
+    ).framework(args.framework).reporting(min_time_s_per_iteration=0.1).training(
+        model={
+            "use_lstm": True,
+            "lstm_cell_size": 32,
+            "lstm_use_prev_action": args.use_prev_action,
+            "lstm_use_prev_reward": args.use_prev_reward,
+        }
+    )
```

Comment:
Can we support here solely the new stack? The reason is that we want new users to jump directly onto the new stack. Here this would mean it needs in addition the following calls to the `config`:

```python
.api_stack(
            enable_rl_module_and_learner=True,
            enable_env_runner_and_connector_v2=True,
)
.rl_module(
            model_config_dict={
                "use_lstm": True,
                "lstm_cell_size": 32,
                "lstm_use_prev_action": args.use_prev_action,
                "lstm_use_prev_reward": args.use_prev_reward,
            }
)
``` 
The `model` argument needs then to vanish from `training()`.

Reply:
Hi @simonsays1980 of course!

I do see some errors occurring when converting over to the new stack. Most likely the observation space, but it might help others down the line for seeing this. Let me know if you have any thoughts, i'll start digging into it!
```
ValueError: The two structures don't have the same nested structure.

First structure: type=dict str={'actor': {'h': array([[0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.]],
      dtype=float32), 'c': array([[0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.]],
      dtype=float32)}, 'critic': {'h': array([[0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.]],
      dtype=float32), 'c': array([[0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.]],
      dtype=float32)}}

Second structure: type=dict str={'h': array([[[-0.05538796,  0.13966578,  0.01491474, ...,  0.02647178,
          0.048148  , -0.07149553]],

       [[ 0.10487001, -0.12613896,  0.1936225 , ..., -0.04218567,
          0.02297537, -0.04598944]],

       [[-0.0181545 ,  0.20065415, -0.03715083, ...,  0.02511652,
          0.0963103 , -0.00859845]],

       ...,

       [[-0.10791852,  0.21027309,  0.04330226, ...,  0.06297404,
         -0.1705255 ,  0.20365342]],

       [[ 0.08050264,  0.1678762 ,  0.14637923, ...,  0.08334337,
         -0.09527908,  0.21276303]],

       [[ 0.08050264,  0.1678762 ,  0.14637923, ...,  0.08334337,
         -0.09527908,  0.21276303]]], dtype=float32), 'c': array([[[-0.10788506,  0.2418374 ,  0.03368664, ...,  0.05303482,
          0.12481229, -0.16512036]],

       [[ 0.2415701 , -0.25722432,  0.3673103 , ..., -0.07641751,
          0.05405141, -0.0850858 ]],

       [[-0.027573  ,  0.2839686 , -0.11287448, ...,  0.05270897,
          0.36143133, -0.02489884]],

       ...,

       [[-0.26991454,  0.49751803,  0.07710288, ...,  0.1402412 ,
         -0.41773236,  0.5066306 ]],

       [[ 0.20296642,  0.47725105,  0.25387886, ...,  0.16696252,
         -0.29451394,  0.40619874]],

       [[ 0.20296642,  0.47725105,  0.25387886, ...,  0.16696252,
         -0.29451394,  0.40619874]]], dtype=float32)}
```

Reply:
Looks like it has to do with the depth of the array between the state_out and the look_back_state.
I am looking to see the differences between the new API and older api. Unsure if this is fixable through an observation wrapper or if it's better to create a new issue. @simonsays1980 Let me know your thoughts!

Reply:
I ruled out it was because LSTM is enabled. Seems to be some problems using it with the new API. I can create another issue for it to be addressed within the new API. Think we're good here for now.

Reply:
@peterghaddad Thanks for digging in. I can replicate this error. Yes this has to do with the LSTM and the connector. I have a guess where this is coming from. We introduced `inference-only` `RLModules` and these do not have a critic. We have to dig into this deeper.

Reply:
@peterghaddad Alright, my guess was right. I fixed it and its in master. Can you merge the newest master pls and check, if the error vanishes?

Reply:
Error still exists unfortunately! Updated the version of Ray by running:
`pip install -U "ray[all] @ https://s3-us-west-2.amazonaws.com/ray-wheels/latest/ray-3.0.0.dev0-cp39-cp39-macosx_10_15_x86_64.whl"`

I think we can move forward without lstm, and I can continue to troubleshoot it on the side. It seems to be the same error we were seeing before.

Reply:
@peterghaddad I cannot see in your fork the new changes - its still behind. Can you pull again from upstream master? From what I can see there are missing a lot of changes - should be also significant faster after the pull - we put a lot into performance and I can see `pyflyt` is quite fast, too :)

Reply:
Yes. Think we're in action now.